src/Integration/src/IntegrationBase/Attributes/ServiceActivatorAttribute.cs

```diff
@@ -28,46 +28,46 @@ public ServiceActivatorAttribute(string inputChannel, string outputChannel, bool
         {
             InputChannel = inputChannel;
             OutputChannel = outputChannel;
-            RequiresReply = requiresReply;
+            RequiresReply = requiresReply.ToString();
         }
 
         public ServiceActivatorAttribute(string inputChannel, string outputChannel, bool requiresReply, int sendTimeout)
         {
             InputChannel = inputChannel;
             OutputChannel = outputChannel;
-            RequiresReply = requiresReply;
-            SendTimeout = sendTimeout;
+            RequiresReply = requiresReply.ToString();
+            SendTimeout = sendTimeout.ToString();
         }
 
         public ServiceActivatorAttribute(string inputChannel, string outputChannel, bool requiresReply, int sendTimeout, bool autoStartup)
         {
             InputChannel = inputChannel;
             OutputChannel = outputChannel;
-            RequiresReply = requiresReply;
-            SendTimeout = sendTimeout;
-            AutoStartup = autoStartup;
+            RequiresReply = requiresReply.ToString();
+            SendTimeout = sendTimeout.ToString();
+            AutoStartup = autoStartup.ToString();
         }
 
         public ServiceActivatorAttribute(string inputChannel, string outputChannel, bool requiresReply, int sendTimeout, bool autoStartup, int phase)
         {
             InputChannel = inputChannel;
             OutputChannel = outputChannel;
-            RequiresReply = requiresReply;
-            SendTimeout = sendTimeout;
-            AutoStartup = autoStartup;
-            Phase = phase;
+            RequiresReply = requiresReply.ToString();
+            SendTimeout = sendTimeout.ToString();
+            AutoStartup = autoStartup.ToString();
+            Phase = phase.ToString();
         }
 
         public string InputChannel { get; set; } = string.Empty;
 
         public string OutputChannel { get; set; } = string.Empty;
 
-        public bool RequiresReply { get; set; } = false;
+        public string RequiresReply { get; set; } = string.Empty;
```

Comment:
Why change all of these to strings?

Reply:
We are using reflection to `ResolveAttribute` and my guess is it somehow easier to resolve to string than bool. In the usage of the attribute itself, it is bool. This is just an implementation detail.  @dtillman - can you confirm?

Reply:
I didn't look at the specific code that references this property, but typically it is a string as a result of allowing users to use `expressions` or `property placeholders` to specify the value of that property ..  Looking at the spring javadocs for ServiceActivator .... (ttps://javadoc.io/static/org.springframework.integration/spring-integration-core/5.0.13.RELEASE/org/springframework/integration/annotation/ServiceActivator.html)  you can see that PropertyPlaceholders are supported on that property .. and probably 'expressions' as well.